experimental/displaynames/src/displaynames.rs

```diff
@@ -75,20 +73,14 @@ impl RegionDisplayNames {
         ]
     );
 
-    /// Returns the locale specific display name of a region for a given string.
-    /// This function is locale-sensitive.
-    pub fn of(&self, code: &str) -> Option<&str> {
-        match <TinyAsciiStr<3>>::from_str(code) {
-            Ok(key) => {
-                let data = self.region_data.get();
-                match self.options.style {
-                    Some(Style::Short) => data.short_names.get(&key),
-                    _ => None,
-                }
-                .or_else(|| data.names.get(&key))
-            }
-            Err(_) => None,
+    /// Returns the display name of a region.
+    pub fn of(&self, region: Region) -> Option<&str> {
```

Comment:
Thought: maybe we should take TinyAsciiStr<3>, or maybe have two versions of this function, one that takes a Region and one that takes an unvalidated string. Also for locales below.

Reply:
+1

Reply:
I'm fine with us taking an `&str` here, `TinyAsciiStr<3>` is `&str`.

Reply:
honestly, I wonder if we should take `&[u8]` more.

Reply:
I don't think we should. It would give a tiny performance gain for the imho uncommon case where you don't have a strongly typed region and can avoid parsing it, but I think usually you'll have a `Region` that you obtained from somewhere, and the API should encourage strong typing.

Reply:
We need `&[u8]` for FFI.

There's basically zero cost to have three APIs:

- `of(StrongType)`
- `of_str(&str)`
- `of_bytes(&[u8])`

Reply:
ok I'm convinced by the normalization argument